group/assignment5/Graph.java

```
@@ -0,0 +1,241 @@
+/**
+ * Assumption : the graph has vertices whose value is a character
+ */
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Stack;
+
+public class Graph {
+  private List<Vertex> vertices;
+
+  public Graph() {
+    vertices = new ArrayList<>();
+  }
+
+  /**
+   * Returns all vertices in the current graph.
+   *
+   * @return the list of the vertices in the current graph
+   */
+  public List<Vertex> getVertices() {
+    return vertices;
+  }
+
+  /**
+   * Adds a vertex to the graph if it does not already exist in the graph.
+   *
+   * @param vertex the vertex to be added.
+   * @return true, if the edge was successfully added, false if it already existed
+   * @throws IllegalArgumentException if the vertex to be added is null.
+   */
+  public boolean addVertex(Vertex vertex) {
+    if (vertex == null) {
```

Nice that you don't just ignore exceptions and put the effort to check them 😄